code/modules/projectiles/guns/revolvers.dm

```diff
@@ -138,6 +127,21 @@
 				to_chat(user, "<span class='warning'>\The [magazine] doesn't fit!</span>")
 		else
 			to_chat(user, "<span class='warning'>You can't load a speedloader when there's something in the cylinder!</span>")
+		return
+
+	// the rest go slow: handfuls, boxes, etc..
+	if(!current_mag.current_rounds && current_mag.caliber == magazine.caliber) //Make sure nothing's loaded and the calibers match.
+		replace_ammo(user, magazine) //We are going to replace the ammo just in case.
+		current_mag.match_ammo(magazine)
+		current_mag.transfer_ammo(magazine,user,1) //Handful can get deleted, so we can't check through it.
+		add_to_cylinder(user)
+	//If bullets still remain in the gun, we want to check if the actual ammo matches.
+	else if(magazine.default_ammo == current_mag.default_ammo) //Ammo datums match, let's see if they are compatible.
+		if(current_mag.transfer_ammo(magazine,user,1))
+			add_to_cylinder(user)//If the magazine is deleted, we're still fine.
+	else
+		to_chat(user, "[current_mag] is [current_mag.current_rounds ? "already loaded with some other ammo. Better not mix them up." : "not compatible with that ammo."]")
```

Comment:
Bwaaah early return

Reply:
I'll update it, literal legacy CM revolver code I just pasta